centaur/src/main/scala/centaur/test/Test.scala

```diff
@@ -647,6 +646,124 @@ object Operations extends StrictLogging {
+              "op" -> JsString("copy"),
+              "from" -> JsString(from.toString),
+              "path" -> JsString(path.toString)))
+            case Move(from, path) => JsObject(Map[String, JsValue](
+              "op" -> JsString("move"),
+              "from" -> JsString(from.toString),
+              "path" -> JsString(path.toString)))
+            case Remove(path, old) => JsObject(Map[String, JsValue](
+              "op" -> JsString("remove"),
+              "path" -> JsString(path.toString)) ++ old.map(o => "old" -> o) )
+            case Replace(path, value, old) => JsObject(Map[String, JsValue](
+              "op" -> JsString("replace"),
+              "path" -> JsString(path.toString),
+              "value" -> value) ++ old.map(o => "old" -> o) )
+            case diffson.jsonpatch.Test(path, value) => JsObject(Map[String, JsValue](
+              "op" -> JsString("test"),
+              "path" -> JsString(path.toString),
+              "value" -> value))
+          }
+
+          override def write(obj: JsonPatch[JsValue]): JsValue = {
+            JsArray(obj.ops.toVector.map(processOperation))
+          }
+        }
+
+        val jsonDiff = diff(expectation: JsValue, actual: JsValue)
+
+        logger.error(s"Bad JM style metadata: ${jsonDiff.toJson.prettyPrint}")
+        IO.raiseError(new Exception(s"Bad JM style metadata. See error log output"))
+      }
+    }
+
+    override def run: IO[Unit] = for {
+      jmMetadata <- CentaurCromwellClient.metadata(workflow = submittedWorkflow, Option(CentaurCromwellClient.defaultMetadataArgs.getOrElse(Map.empty) ++ jmArgs))
+      jmMetadataObject <- IO.fromTry(Try(jmMetadata.value.parseJson.asJsObject))
+      expectation <- IO.fromTry(Try(setUpJmStyleMetadataExpectation(originalMetadata.parseJson.asJsObject)))
+
+      validationUnit <- validate(expectation, jmMetadataObject)
+    } yield validationUnit
+  }
+
+  val oneWordIncludeKeys = List(
+    "attempt", "callRoot", "end",
+    "executionStatus", "failures", "inputs", "jobId",
+    "calls", "outputs", "shardIndex", "start", "stderr", "stdout",
+    "description", "executionEvents", "labels", "parentWorkflowId",
+    "returnCode", "status", "submission", "subWorkflowId", "workflowName"
+  )
+
+  val jmArgs = Map(
+    "includeKey" -> (oneWordIncludeKeys :+ "callCaching:hit"),
+    "excludeKey" -> List("callCaching:hitFailures"),
+    "expandSubWorkflows" -> List("false")
+  )
+
+  // I chose to re-implement "process metadata for job manager's includeKeys/excludeKeys" and not re-use production.
+  // Reasoning is:
+  //  1. the ultra-specific JM case is a lot simpler to reason about and can be done in a few lines
+  //  2. it's nice to be able to assert general cases rather than singular specific examples
+  //  3. it's hopefully unlikely that the same bug will show up in two separate implementations
+  def setUpJmStyleMetadataExpectation(originalWorkflowMetadataJson: JsObject): JsObject = {
```

Comment:
TOL: Method name seems a bit confusing to me. Wouldn't it be better to call it something like `convertOriginalMetadataIntoJmStyleMetadata` or `composeJmStyleMetadataFromOriginalMetadata`?

Reply:
Sure, that works. I'll make the change